pkg/dns/aws/client.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2022 The MultiCluster Traffic Controller Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/service/route53"
+)
+
+type InstrumentedRoute53 struct {
+	route53 *route53.Route53
+}
+
+func observe(operation string, f func() error) {
+	_ = f()
+	//ToDo metrics
+	//start := time.Now()
```

Comment:
Is there a reason not to have these metrics now?

Reply:
Added metrcs